lawnchair/src/app/lawnchair/search/SearchTargetUtils.kt

```diff
@@ -7,14 +7,23 @@ import androidx.core.os.bundleOf
 import com.android.app.search.LayoutType
 import com.android.launcher3.model.data.AppInfo
 import com.android.launcher3.util.ComponentKey
+import java.security.MessageDigest
+
+// We're generate hash key as alt in id, so we can avoid
+// the wrong action when click,
+// TODO remove when we properly manage creating search target
+private fun generateHashKey(input: String): String =
+    MessageDigest.getInstance("SHA-256")
+        .digest(input.toByteArray())
+        .joinToString { "%02x".format(it) }
```

Comment:
you can do this easily with Okio.

Reply:
is this fine? `ByteString.of(*MessageDigest.getInstance("SHA-256").digest(input.toByteArray`

Reply:
Something like:
```kt
import okio.ByteString
import okio.ByteString.Companion.toByteString

fun ByteArray.sha1(separator: CharSequence = ""): String = toByteString().sha1().toHex(separator)

fun ByteArray.sha256(separator: CharSequence = ""): String =
  toByteString().sha256().toHex(separator)

fun ByteArray.md5(separator: CharSequence = ""): String = toByteString().md5().toHex(separator)

fun ByteArray.toHexString(separator: CharSequence = ""): String = joinToString(separator) {
  it.toInt().and(0xff).toString(16).padStart(2, '0')
}

private fun ByteString.toHex(separator: CharSequence) = toByteArray().toHexString(separator)
```

Reply:
This is nice, but the current kinda fast, can you confirm how's this better the current one?

Reply:
Little performance difference, but easy to maintain.

Reply:
Right, I think we can do that when we have other use cases for now let's keep `ByteString.of(*MessageDigest.getInstance("SHA-256").digest(input.toByteArray`